cmake/tracy.cmake

```diff
@@ -28,7 +28,7 @@ ExternalProject_Add(
     INSTALL_COMMAND
         cp ${TRACY_HOME}/csvexport/build/unix/csvexport-release .
     BUILD_COMMAND
-        cd ${TRACY_HOME}/csvexport/build/unix && TRACY_NO_LTO=1 make -f ${TRACY_HOME}/csvexport/build/unix/Makefile
+        cd ${TRACY_HOME}/csvexport/build/unix && CXX=g++ TRACY_NO_LTO=1 make -f ${TRACY_HOME}/csvexport/build/unix/Makefile
```

Comment:
@mo-tenstorrent @vtangTT 
Are we aware of this? This seems like a notable change

Reply:
this will not go in from my side :) i will remove it from this PR.

Reply:
I have an old PR [here](https://github.com/tenstorrent/tt-metal/pull/9425), but @TT-billteng and I didn't want to force tracy to use g++ while we are using Clang for metal. 

The current plan is the bump Tracy tools to use CMakeLists ([issue](https://github.com/tenstorrent/tt-metal/issues/9691))so we have better control over the build env. Encountering a number of build problems but I will be actively working on this these next few days and hope to get it in by end of week. 

I don't want to block your [PR](https://github.com/tenstorrent/tt-metal/pull/9660), so we could submit the workaround for now? @TT-billteng

Reply:
@mywoodstock without this your builds break on ird right?

Reply:
Yes, unable to build on IRD without this hack.
@vtangTT @TT-billteng if this workaround is ok, i can keep it with this PR. LMK.

Reply:
let's submit this for now and try to fix it with the Tracy submodule bump